pkg/model/client_secret.go

```diff
@@ -1,16 +1,35 @@
 package model
 
 import (
+	"unicode"
+
 	"github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
 	v1 "k8s.io/api/core/v1"
 	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func MakeK8sCompatibleName(text string) string {
+	// we only want letters and numbers
+	reg := []rune(SanitizeResourceName(text))
+
+	// we guarantee first and last char is an alpha and not a dot or a dash
+	if !unicode.IsLetter(reg[0]) && !unicode.IsNumber(reg[0]) {
+		reg = append([]rune{'a'}, reg...)
+	}
+	if !unicode.IsLetter(reg[len(reg)-1]) && !unicode.IsNumber(reg[len(reg)-1]) {
+		reg = append(reg, 'a')
+	}
+
+	return string(reg)
+}
```

Comment:
Shouldn't we move this to the util file and rename it to something like `SanitizeResourceNameWithAlphaNum`? Not sure about the name, though...

Reply:
OK